spring-integration-gcp/src/main/java/org/springframework/integration/gcp/outbound/PubSubMessageHandler.java

```diff
@@ -29,15 +41,60 @@
 
 	private final PubSubOperations pubSubTemplate;
 
+	private MessageConverter messageConverter;
+
 	private String topic;
 
 	public PubSubMessageHandler(PubSubOperations pubSubTemplate, String topic) {
 		this.pubSubTemplate = pubSubTemplate;
 		this.topic = topic;
+
+		StringMessageConverter stringMessageConverter = new StringMessageConverter();
+		stringMessageConverter.setSerializedPayloadClass(String.class);
+		this.messageConverter =
+				new CompositeMessageConverter(ImmutableList.of(stringMessageConverter));
 	}
 
 	@Override
-	protected void handleMessageInternal(Message<?> message) throws Exception {
-		this.pubSubTemplate.publish(this.topic, message);
+	protected void handleMessageInternal(Message<?> message) {
+		Object payload = message.getPayload();
+
+		if (payload instanceof PubsubMessage) {
+			this.pubSubTemplate.publish(this.topic, (PubsubMessage) payload);
+			return;
+		}
+
+		ByteString pubsubPayload;
+
+		if (payload instanceof String) {
```

Comment:
@artembilan what do you think about this?

I looked at other integration outbound adapters, but they all followed a different approach.

Reply:
See my comment on the exception: https://github.com/spring-cloud/spring-cloud-gcp/pull/151#discussion_r140080119